src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,107 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final int ID_INDEX = 1;
+    private static final int ID_POSITION = 3;
+    private static final int PRICE_POSITION = 2;
+    private static final int TITLE_POSITION = 1;
+    private static final String INSERT = "INSERT INTO books (title, price) VALUES (?, ?)";
```

Comment:
You can store the queries as local variables, it'll make your code easier to maintain.